src/couch_replicator_httpc.erl

```diff
@@ -226,21 +234,45 @@ clean_mailbox(_, Count) when Count > 0 ->
     ok.
 
 
+%% For 429 errors, we perform an exponential backoff up to 2.17 hours.
+%% We use Backoff time as a timeout/failure end.
+maybe_retry(backoff, Worker, #httpdb{backoff = Backoff} = HttpDb, Params) ->
+    ok = timer:sleep(random:uniform(Backoff)),
+    Backoff2 = round(Backoff*?BACKOFF_EXP),
+    NewBackoff= erlang:min(Backoff2, ?MAX_BACKOFF_WAIT),
+    NewHttpDb = HttpDb#httpdb{backoff = NewBackoff},
+    case Backoff2 of
+        W0 when W0 >= ?MAX_BACKOFF_LOG_THRESHOLD -> % Past 8 min, we log retries
+            log_retry_error(Params, HttpDb, Backoff2, "429 Retry"),
+            throw({retry, NewHttpDb, Params});
+        W1 when W1 > ?MAX_BACKOFF_WAIT ->
+            report_error(Worker, HttpDb, Params, {error,
+                "429 Retry Timeout"});
+        _ ->
+            throw({retry, NewHttpDb, Params})
+    end;
+
 maybe_retry(Error, Worker, #httpdb{retries = 0} = HttpDb, Params) ->
     report_error(Worker, HttpDb, Params, {error, Error});
 
 maybe_retry(Error, _Worker, #httpdb{retries = Retries, wait = Wait} = HttpDb,
     Params) ->
-    Method = string:to_upper(atom_to_list(get_value(method, Params, get))),
-    Url = couch_util:url_strip_password(full_url(HttpDb, Params)),
-    twig:log(notice,"Retrying ~s request to ~s in ~p seconds due to error ~s",
-        [Method, Url, Wait / 1000, error_cause(Error)]),
-    ok = timer:sleep(Wait),
+    log_retry_error(Params, HttpDb, Wait, Error),
+    % This is so that a long backoff time is not used to ensure
+    % backwards compatibility.
+    ok = timer:sleep(erlang:min(Wait, ?MAX_WAIT)),
```

Comment:
only the extraction of log_retry_error is needed here now that backoff is its own field?